Lidgren.Network/NetBuffer.cs

```
@@ -14,14 +14,16 @@ public partial class NetBuffer
 		private static readonly Dictionary<Type, MethodInfo> s_readMethods;
 		private static readonly Dictionary<Type, MethodInfo> s_writeMethods;
 
-		internal byte[] m_data;
+		internal byte[] BufferData => m_data ?? throw new InvalidOperationException($"{nameof(m_data)} is null");
+
+		internal byte[]? m_data;
 		internal int m_bitLength;
 		internal int m_readPosition;
 
 		/// <summary>
 		/// Gets or sets the internal data buffer
 		/// </summary>
-		public byte[] Data
+		public byte[]? Data

```

As far as I can tell, `m_data` can only be null if the message is currently pooled. I'd just make `Data` throw in that case (as it's kind of an invalid state to be doing stuff with it, you shouldn't have a handle) so it doesn't need to be nullable.